From d4d4973c8931fd1b0f39139050734cc835da5731 Mon Sep 17 00:00:00 2001
From: Simo Sorce <simo@redhat.com>
Date: Thu, 17 Jul 2025 09:40:34 -0400
Subject: [PATCH 55/58] Add a define to disable symver attributes

Defininig RHEL_NO_SYMVER_ATTRIBUTES for a build now prevents adding
compatibility symver attributes.

Signed-off-by: Simo Sorce <simo@redhat.com>
---
 crypto/evp/digest.c  | 2 +-
 crypto/evp/evp_enc.c | 2 +-
 crypto/o_str.c       | 4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/crypto/evp/digest.c b/crypto/evp/digest.c
index 101f49862c..dbf9bbba7a 100644
--- a/crypto/evp/digest.c
+++ b/crypto/evp/digest.c
@@ -573,7 +573,7 @@ int EVP_DigestSqueeze(EVP_MD_CTX *ctx, unsigned char *md, size_t size)
 }
 
 EVP_MD_CTX
-#if !defined(FIPS_MODULE) && !defined(OPENSSL_SYS_UEFI)
+#if !defined(FIPS_MODULE) && !defined(OPENSSL_SYS_UEFI) && !defined(RHEL_NO_SYMVER_ATTRIBUTES)
 __attribute__ ((symver ("EVP_MD_CTX_dup@@OPENSSL_3.1.0"),
                     symver ("EVP_MD_CTX_dup@OPENSSL_3.2.0")))
 #endif
diff --git a/crypto/evp/evp_enc.c b/crypto/evp/evp_enc.c
index 30c2ec56dc..4117fafdfa 100644
--- a/crypto/evp/evp_enc.c
+++ b/crypto/evp/evp_enc.c
@@ -1767,7 +1767,7 @@ int EVP_CIPHER_CTX_rand_key(EVP_CIPHER_CTX *ctx, unsigned char *key)
 }
 
 EVP_CIPHER_CTX
-#if !defined(FIPS_MODULE) && !defined(OPENSSL_SYS_UEFI)
+#if !defined(FIPS_MODULE) && !defined(OPENSSL_SYS_UEFI) && !defined(RHEL_NO_SYMVER_ATTRIBUTES)
 __attribute__ ((symver ("EVP_CIPHER_CTX_dup@@OPENSSL_3.1.0"),
                     symver ("EVP_CIPHER_CTX_dup@OPENSSL_3.2.0")))
 #endif
diff --git a/crypto/o_str.c b/crypto/o_str.c
index b221ba789e..ca583d9ce6 100644
--- a/crypto/o_str.c
+++ b/crypto/o_str.c
@@ -418,7 +418,7 @@ int openssl_strerror_r(int errnum, char *buf, size_t buflen)
 }
 
 int
-#if !defined(FIPS_MODULE) && !defined(OPENSSL_SYS_UEFI)
+#if !defined(FIPS_MODULE) && !defined(OPENSSL_SYS_UEFI) && !defined(RHEL_NO_SYMVER_ATTRIBUTES)
 __attribute__ ((symver ("OPENSSL_strcasecmp@@OPENSSL_3.0.3"),
                     symver ("OPENSSL_strcasecmp@OPENSSL_3.0.1")))
 #endif
@@ -433,7 +433,7 @@ OPENSSL_strcasecmp(const char *s1, const char *s2)
 }
 
 int
-#if !defined(FIPS_MODULE) && !defined(OPENSSL_SYS_UEFI)
+#if !defined(FIPS_MODULE) && !defined(OPENSSL_SYS_UEFI) && !defined(RHEL_NO_SYMVER_ATTRIBUTES)
 __attribute__ ((symver ("OPENSSL_strncasecmp@@OPENSSL_3.0.3"),
                     symver ("OPENSSL_strncasecmp@OPENSSL_3.0.1")))
 #endif
-- 
2.55.0

